### ISSUE TITLE
Issues with dynamic revisions

### ISSUE THREAD

**Issue by rockjam** [alexarchambault/coursier#104]
Found some issues with resolution, when ivy dynamic revisions are present.

Unabled to resolve dependency declared in following way:
`"com.googlecode.libphonenumber" % "libphonenumber"                 % "7.0.+"`

with following stack trace:
```
1 error(s):
  com.googlecode.libphonenumber:libphonenumber:7.0.+:
    /Users/rockjam/.sbt/0.13/plugins/target/resolution-cache/com.googlecode.libphonenumber/libphonenumber/7.0.+/resolved.xml.xml: not found
    Not found
    /Users/rockjam/.ivy2/local/com.googlecode.libphonenumber/libphonenumber/7.0.+/ivys/ivy.xml: not found
    Download error: Caught java.io.FileNotFoundException: https://repo.jfrog.org/artifactory/libs-release-bintray/com/googlecode/libphonenumber/libphonenumber/7.0.%2B/libphonenumber-7.0.%2B.pom?referrer (https://repo.jfrog.org/artifactory/libs-release-bintray/com/googlecode/libphonenumber/libphonenumber/7.0.%2B/libphonenumber-7.0.%2B.pom?referrer)
    Download error: Caught java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://oss.sonatype.org/content/repositories/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://oss.sonatype.org/content/repositories/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/scalaz/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/scalaz/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/dnvriend/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/dnvriend/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/dwhjames/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/dwhjames/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/krasserm/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/krasserm/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://raw.githubusercontent.com/slorber/gcm-server-repository/master/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://raw.githubusercontent.com/slorber/gcm-server-repository/master/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://oss.sonatype.org/content/repositories/snapshots/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://oss.sonatype.org/content/repositories/snapshots/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://repo1.maven.org/maven2/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://oss.sonatype.org/content/repositories/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://oss.sonatype.org/content/repositories/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/scalaz/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/scalaz/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/dnvriend/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/dnvriend/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/dwhjames/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/dwhjames/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://dl.bintray.com/krasserm/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://dl.bintray.com/krasserm/maven/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://raw.githubusercontent.com/slorber/gcm-server-repository/master/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://raw.githubusercontent.com/slorber/gcm-server-repository/master/releases/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
    Download error: Caught java.io.FileNotFoundException: https://oss.sonatype.org/content/repositories/snapshots/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom (https://oss.sonatype.org/content/repositories/snapshots/com/googlecode/libphonenumber/libphonenumber/7.0.+/libphonenumber-7.0.+.pom)
```
And I got one more error after I changed 7.0.+ to specific version:

```
java.lang.Exception: Exception during resolution
	at coursier.Tasks$$anonfun$updateTask$1$$anonfun$13.apply(Tasks.scala:282)
	at coursier.Tasks$$anonfun$updateTask$1$$anonfun$13.apply(Tasks.scala:282)
	at scalaz.$bslash$div.leftMap(Either.scala:97)
	at coursier.Tasks$$anonfun$updateTask$1.coursier$Tasks$$anonfun$$report$1(Tasks.scala:282)
	at coursier.Tasks$$anonfun$updateTask$1$$anonfun$apply$15.apply(Tasks.scala:431)
	at coursier.Tasks$$anonfun$updateTask$1$$anonfun$apply$15.apply(Tasks.scala:431)
	at scala.collection.mutable.MapLike$class.getOrElseUpdate(MapLike.scala:189)
	at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:91)
	at coursier.Tasks$$anonfun$updateTask$1.apply(Tasks.scala:429)
	at coursier.Tasks$$anonfun$updateTask$1.apply(Tasks.scala:120)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.AssertionError: assertion failed: URL: Users/rockjam/.sbt/0.13/plugins/target/resolution-cache/org.yaml/snakeyaml/[1.9,2.0)/resolved.xml.xml, file: Users/rockjam/.sbt/0.13/plugins/target/resolution-cache/org.yaml/snakeyaml/%5B1.9,2.0)/resolved.xml.xml
	at scala.Predef$.assert(Predef.scala:179)
	at coursier.Cache$$anonfun$11.apply(Cache.scala:306)
	at coursier.Cache$$anonfun$11.apply(Cache.scala:299)
	at scala.collection.TraversableLike$WithFilter$$anonfun$map$2.apply(TraversableLike.scala:722)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:721)
	at coursier.Cache$.download(Cache.scala:299)
	at coursier.Cache$.file(Cache.scala:410)
	at coursier.Cache$$anonfun$fetch$1.apply(Cache.scala:456)
	at coursier.Cache$$anonfun$fetch$1.apply(Cache.scala:455)
	at coursier.ivy.IvyRepository$$anonfun$find$2.apply(IvyRepository.scala:261)
	at coursier.ivy.IvyRepository$$anonfun$find$2.apply(IvyRepository.scala:260)
	at scalaz.EitherT$$anonfun$flatMap$1$$anonfun$apply$4.apply(EitherT.scala:91)
	at scalaz.$bslash$div.fold(Either.scala:57)
	at scalaz.EitherT$$anonfun$flatMap$1.apply(EitherT.scala:91)
	at scalaz.EitherT$$anonfun$flatMap$1.apply(EitherT.scala:91)
	at scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:35)
	at scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:35)
	at scalaz.concurrent.Task$.Try(Task.scala:386)
	at scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:35)
	at scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:33)
	at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
	at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
	at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
	at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
	at scalaz.concurrent.Future.step(Future.scala:109)
	at scalaz.concurrent.Future.listen(Future.scala:75)
	at scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2$$anonfun$apply$10.apply(Future.scala:309)
	at scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2$$anonfun$apply$10.apply(Future.scala:308)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2.apply(Future.scala:308)
	at scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2.apply(Future.scala:304)
	at scalaz.concurrent.Future.listen(Future.scala:79)
	at scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
	at scalaz.Free.resume(Free.scala:73)
	at scalaz.Free.go2$1(Free.scala:118)
	at scalaz.Free.go(Free.scala:122)
	at scalaz.Free.run(Free.scala:172)
	at scalaz.concurrent.Future.listen(Future.scala:76)
	at scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
	at scalaz.Free.resume(Free.scala:73)
	at scalaz.Free.go2$1(Free.scala:118)
	at scalaz.Free.go(Free.scala:122)
	at scalaz.Free.run(Free.scala:172)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$4.call(Future.scala:380)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$4.call(Future.scala:380)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
As I can see, that stack trace points to Cache.scala: 306, which was removed in this revision: https://github.com/alexarchambault/coursier/commit/214e0f893efd6f1aebaebebbc49ce441944c5490#diff-865e8f99a3d136acc3acb4ffcdd0b626L306 

So, I guess it will be fixed with next release

**Comment by alexarchambault** [comment]
About the first point: the syntax for Ivy version range is not supported yet, it should be easy to add though, as the Maven one is fine (replacing `7.0.+` with `[7.0,7.1)` works for me - it returns the `7.0.11` as expected).

About the second point, indeed, it's fixed in master.

**Comment by rockjam** [comment]
Ok, I'll try again with local snapshot. How soon do you expect to make next release?

**Comment by alexarchambault** [comment]
`1.0.0-M4` is released and has a fix for the second point. (cc @andrewconner)

**Comment by rockjam** [comment]
Thank you. Second one is fixed in `1.0.0-M4`
